core/project_settings.cpp

```diff
@@ -525,6 +525,8 @@ Error ProjectSettings::_load_settings_binary(const String &p_path) {
 		set(key, value);
 	}
 
+	f->close();
```

Comment:
Do we really need to `close` before `memdelete`? Because if so there are tons of locations where this is not done currently and we just `memdelete(f)`.